src/output-json.c

```diff
@@ -724,6 +724,41 @@ json_t *CreateJSONHeader(const Packet *p, enum OutputJsonLogDirection dir,
 
     CreateJSONFlowId(js, f);
 
+    if (p->ethh && (options_flags & LOGFILE_LOG_ETHERNET)) {
```

Comment:
I'd like to see
1. Utility routine (e.g., `CreateJSONEther`) used here and from `output-json-flow.c`. `JSONFlowAddMACAddrs` looks like a good start at a utility routine.
2. An inline function to `sprintf` and then `json_object_set_new` the formatted mac string.